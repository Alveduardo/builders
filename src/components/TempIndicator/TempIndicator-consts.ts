export const TEMP_INDICATOR = {
  TEST_ID: {
    CONTAINER: 'TEMP_INDICATOR_CONTAINER_TEST_ID',
    GRADIENT: 'TEMP_INDICATOR_GRADIENT_TEST_ID',
    MIN: 'TEMP_INDICATOR_MIN_TEST_ID',
    MAX: 'TEMP_INDICATOR_MAX_TEST_ID',
  },
  GRADIENT: {
    START: { x: 0, y: 0 },
    END: { x: 1, y: 0 },
  },
};
