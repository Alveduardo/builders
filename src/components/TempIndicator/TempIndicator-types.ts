import { ViewProps, ViewStyle } from 'react-native';

interface DefaultStyles {
  containerStyle: ViewStyle;
  gradientStyle: ViewStyle;
}

export interface TempIndicatorStyles {
  defaultStyles: DefaultStyles;
}

interface StylesConfig extends DefaultStyles {}

export interface TempIndicatorConfig {
  styles: StylesConfig;
}

export interface TempIndicatorProps extends ViewProps {
  min: number;
  max: number;
}
