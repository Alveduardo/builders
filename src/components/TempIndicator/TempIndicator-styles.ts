import { TempIndicatorStyles } from './TempIndicator-types';

export const tempIndicatorStyles: TempIndicatorStyles = {
  defaultStyles: {
    containerStyle: {
      alignItems: 'center',
      flexDirection: 'row',
    },
    gradientStyle: {
      width: 80,
      height: 6,
      borderRadius: 999,
      marginHorizontal: 8,
    },
  },
};
