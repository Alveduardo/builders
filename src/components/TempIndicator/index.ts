import TempIndicator from './TempIndicator';
import { TempIndicatorProps } from './TempIndicator-types';

export default TempIndicator;
export { TempIndicatorProps };
