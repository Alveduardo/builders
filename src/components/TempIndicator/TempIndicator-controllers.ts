import { ViewStyle } from 'react-native';
import { TEMP_INDICATOR } from './TempIndicator-consts';
import { TempIndicatorProps } from './TempIndicator-types';
import { tempIndicatorStyles } from './TempIndicator-styles';

import { setupStyleProp } from '../../utils/style/style-utils';

export const useTempIndicatorConfig = ({ style }: Partial<TempIndicatorProps>) => {
  let {
    defaultStyles: { containerStyle, gradientStyle },
  } = tempIndicatorStyles;

  containerStyle = setupStyleProp(style, containerStyle, TEMP_INDICATOR.ALLOWED_PROPERTIES) as ViewStyle;

  return {
    styles: {
      containerStyle,
      gradientStyle,
    },
  };
};
