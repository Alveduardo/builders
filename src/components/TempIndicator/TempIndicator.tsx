import React from 'react';
import { View } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import { TEMP_INDICATOR } from './TempIndicator-consts';
import { TempIndicatorProps } from './TempIndicator-types';
import { useTempIndicatorConfig } from './TempIndicator-controllers';

import Label from '../Label';

const TempIndicator = ({ min, max, style, ...rest }: TempIndicatorProps): JSX.Element => {
  const {
    styles: { containerStyle, gradientStyle },
  } = useTempIndicatorConfig({ style });

  return (
    <View testID={TEMP_INDICATOR.TEST_ID.CONTAINER} {...rest} {...{ style: [containerStyle, style] }}>
      <Label testID={TEMP_INDICATOR.TEST_ID.MIN}>{min}°</Label>
      <LinearGradient
        style={gradientStyle}
        testID={TEMP_INDICATOR.TEST_ID.GRADIENT}
        colors={TEMP_INDICATOR.GRADIENT.COLORS}
        start={TEMP_INDICATOR.GRADIENT.START}
        end={TEMP_INDICATOR.GRADIENT.END}
      />
      <Label testID={TEMP_INDICATOR.TEST_ID.MAX}>{max}°</Label>
    </View>
  );
};

export default TempIndicator;
