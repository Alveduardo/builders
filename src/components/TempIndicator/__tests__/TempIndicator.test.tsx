import React from 'react';
import { render } from '@testing-library/react-native';

import TempIndicator from '../TempIndicator';
import { TEMP_INDICATOR } from '../TempIndicator-consts';

const tempIndicatorPropsDefault = {
  min: 0,
  max: 0,
};

jest.mock('react-native-linear-gradient', () => 'LinearGradient');

describe('TempIndicator ->', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(jest.fn());
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });

  describe('Snapshot ->', () => {
    it('Deve corresponder ao componente com as props default', () => {
      const tree = render(<TempIndicator {...tempIndicatorPropsDefault} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Deve corresponder ao componente com as props mais utilizadas', () => {
      const tree = render(<TempIndicator {...tempIndicatorPropsDefault} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('props ->', () => {
    describe('min ->', () => {
      it('Deve renderizar o componente exibindo a prop min', () => {
        const { getByTestId } = render(<TempIndicator {...tempIndicatorPropsDefault} />);

        const component = getByTestId(TEMP_INDICATOR.TEST_ID.MIN);

        expect(component.children[0]).toBe(tempIndicatorPropsDefault.min.toString());
      });
    });
    describe('max ->', () => {
      it('Deve renderizar o componente exibindo a prop max', () => {
        const { getByTestId } = render(<TempIndicator {...tempIndicatorPropsDefault} />);

        const component = getByTestId(TEMP_INDICATOR.TEST_ID.MIN);

        expect(component.children[0]).toBe(tempIndicatorPropsDefault.max.toString());
      });
    });
  });
});
