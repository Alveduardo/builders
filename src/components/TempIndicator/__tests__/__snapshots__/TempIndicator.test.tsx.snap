// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TempIndicator -> Snapshot -> Deve corresponder ao componente com as props default 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  testID="TEMP_INDICATOR_CONTAINER_TEST_ID"
>
  <Text
    rest={
      Object {
        "testID": "TEMP_INDICATOR_MIN_TEST_ID",
      }
    }
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
        undefined,
      ]
    }
    testID="TEMP_INDICATOR_MIN_TEST_ID"
  >
    0
    째
  </Text>
  <LinearGradient
    colors={
      Array [
        "#b6f29b",
        "#ff6a38",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    start={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "borderRadius": 999,
        "height": 6,
        "marginHorizontal": 8,
        "width": 80,
      }
    }
    testID="TEMP_INDICATOR_GRADIENT_TEST_ID"
  />
  <Text
    rest={
      Object {
        "testID": "TEMP_INDICATOR_MAX_TEST_ID",
      }
    }
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
        undefined,
      ]
    }
    testID="TEMP_INDICATOR_MAX_TEST_ID"
  >
    0
    째
  </Text>
</View>
`;

exports[`TempIndicator -> Snapshot -> Deve corresponder ao componente com as props mais utilizadas 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  testID="TEMP_INDICATOR_CONTAINER_TEST_ID"
>
  <Text
    rest={
      Object {
        "testID": "TEMP_INDICATOR_MIN_TEST_ID",
      }
    }
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
        undefined,
      ]
    }
    testID="TEMP_INDICATOR_MIN_TEST_ID"
  >
    0
    째
  </Text>
  <LinearGradient
    colors={
      Array [
        "#b6f29b",
        "#ff6a38",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    start={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "borderRadius": 999,
        "height": 6,
        "marginHorizontal": 8,
        "width": 80,
      }
    }
    testID="TEMP_INDICATOR_GRADIENT_TEST_ID"
  />
  <Text
    rest={
      Object {
        "testID": "TEMP_INDICATOR_MAX_TEST_ID",
      }
    }
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
        undefined,
      ]
    }
    testID="TEMP_INDICATOR_MAX_TEST_ID"
  >
    0
    째
  </Text>
</View>
`;
